### PR TITLE
test(all): e2e (part 1- shard sequencer test)

### PR DESCRIPTION
**Pull request by technicallyty**
Related to: WORLD-96

## What is the purpose of the change

adds a simple cardinal game shard and a end to end test to ensure the transactions are being stored on chain.

## Brief Changelog

- add simple game shard
- add e2e test

## Testing and Verifying

run `make game` and `make rollup`

remove the t.Skip and then run the test once both stacks are running.

### CODE REVIEW

File: chain/scripts/start-rollup.sh [hunk 1]
```diff
@@ -47,4 +47,4 @@ sed -i '/api\]/,/\[/ s/enable = false/enable = true/' /root/.world/config/app.to
```
Comment (heronimus) [inline chain/scripts/start-rollup.sh:47]:
```suggestion

# Cosmos SDK gRPC listener
sed -i'.bak' 's#"localhost:9090"#"0.0.0.0:9090"#g' /root/.world/config/app.toml
# Cosmos SDK API server listener
sed -i'.bak' 's#"localhost:1317"#"0.0.0.0:1317"#g' /root/.world/config/app.toml

```

File: chain/scripts/start-rollup.sh [hunk 2]
```diff
@@ -45,6 +45,11 @@ sed -i'.bak' 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:26657"#g' /root/.world/co
 
 sed -i '/api\]/,/\[/ s/enable = false/enable = true/' /root/.world/config/app.toml
 
+# Cosmos SDK gRPC listener
+sed -i'.bak' 's#"localhost:9090"#"0.0.0.0:9090"#g' /root/.world/config/app.toml
+# Cosmos SDK API server listener
+sed -i'.bak' 's#"localhost:1317"#"0.0.0.0:1317"#g' /root/.world/config/app.toml
```
Comment (heronimus) [inline chain/scripts/start-rollup.sh:51]: I put an extra double quote here before; just realized it when I deployed it previously.
```suggestion
sed -i'.bak' 's#localhost:1317#0.0.0.0:1317#g' /root/.world/config/app.toml
```

**Review by pyrofolium — CHANGES_REQUESTED** [review]
I feel the infra for e2e tests here are redundant to `mage test` in starter-game-template. 
It's unfortunate no one noticed that there was a parallel effort here. 

I think all the tests need to be combined into one repo (this one). And all tests run with one command. So all regular world-engine tests, all starter-game-template tests, and all of the new e2e you created should be run here. 

And it should be either mage or make, I don't think we should have both. I'm not too deeply familiar with either mage or make but given that are whole ecosystem is golang anyway I would prefer mage to keep things consistent and simpler (as I don't have to learn make)

I'm going to place this under "request changes" but if you want to punt this to another issue, lmk I can approve. I'd punt it too. But I feel this overall issue is something that must be done eventually so we should talk about it a little get some consensus on the overall direction here. 

As we talked about earlier there were failing tests on the starter-game-template side so getting these combined improves overall robustness


These are my thoughts on the matter let me know yours.

**Comment by technicallyty** [comment]
> I feel the infra for e2e tests here are redundant to `mage test` in starter-game-template. It's unfortunate no one noticed that there was a parallel effort here.
> 
> I think all the tests need to be combined into one repo (this one). And all tests run with one command. So all regular world-engine tests, all starter-game-template tests, and all of the new e2e you created should be run here.
> 
> And it should be either mage or make, I don't think we should have both. I'm not too deeply familiar with either mage or make but given that are whole ecosystem is golang anyway I would prefer mage to keep things consistent and simpler (as I don't have to learn make)
> 
> I'm going to place this under "request changes" but if you want to punt this to another issue, lmk I can approve. I'd punt it too. But I feel this overall issue is something that must be done eventually so we should talk about it a little get some consensus on the overall direction here.
> 
> As we talked about earlier there were failing tests on the starter-game-template side so getting these combined improves overall robustness
> 
> These are my thoughts on the matter let me know yours.

this e2e test is for chain related testing. it is different than the one in starter game template

**Comment by pyrofolium** [comment]
I recall everyone agreeing on using mage? 

Either way moving 

> > I feel the infra for e2e tests here are redundant to `mage test` in starter-game-template. It's unfortunate no one noticed that there was a parallel effort here.
> > I think all the tests need to be combined into one repo (this one). And all tests run with one command. So all regular world-engine tests, all starter-game-template tests, and all of the new e2e you created should be run here.
> > And it should be either mage or make, I don't think we should have both. I'm not too deeply familiar with either mage or make but given that are whole ecosystem is golang anyway I would prefer mage to keep things consistent and simpler (as I don't have to learn make)
> > I'm going to place this under "request changes" but if you want to punt this to another issue, lmk I can approve. I'd punt it too. But I feel this overall issue is something that must be done eventually so we should talk about it a little get some consensus on the overall direction here.
> > As we talked about earlier there were failing tests on the starter-game-template side so getting these combined improves overall robustness
> > These are my thoughts on the matter let me know yours.
> 
> this e2e test is for chain related testing. it is different than the one in starter game template

As discussed these e2e tests are just for testing chain related stuff. For the future we need to combine all tests together.

**Review by namrapatel — APPROVED** [review]

**Pull request merged**